Source/JavaScriptCore/dfg/DFGConstantFoldingPhase.cpp

```diff
@@ -475,6 +475,31 @@ class ConstantFoldingPhase : public Phase {
                 break;
             }
 
+            case CheckDetached: {
+                AbstractValue& abstractValue = m_state.forNode(node->child1());
+                if (abstractValue.m_type && abstractValue.isType(SpecObject) && abstractValue.m_structure.isFinite()) {
+                    bool canFold = !abstractValue.m_structure.isClear();
+                    JSGlobalObject* globalObject = m_graph.globalObjectFor(node->origin.semantic);
+                    abstractValue.m_structure.forEach([&](RegisteredStructure structure) {
+                        if (structure->globalObject() != globalObject) {
+                            canFold = false;
+                            return;
```

Comment:
I think this `return` is not needed, right?

Reply:
I think anyway it is good when we consider adding the other conditions